src/layer.rs

```diff
@@ -1,102 +1,732 @@
-use serde::Serialize;
+use std::fmt;
 
-#[derive(Debug, Clone, Default, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Layer {
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::{closure::Closure, JsValue};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Expression<T> {
+    Enum(T),
+    String(String),
+    Number(f64),
+    Bool(bool),
+    List(Vec<Expression<T>>),
+}
```

Comment:
This has basically replaced:
```rust
/// Layout property can be either value in String or Number e.g. 0.25
/// or tuple of 2 elements, ("get", "icon").
#[derive(Debug, Clone, Serialize)]
#[serde(untagged)]
pub enum LayoutProperty {
    String(String),
    Number(f64),
    FeatureProperty((String, String)),
}
```

Since all of these 'feature properties' can actually be full blown [mapbox expressions](https://docs.mapbox.com/style-spec/reference/expressions/). I decided to lean in and just kinda add types for a [Lisp](https://stopa.io/post/222) since that's what the expressions are! I considered trying to make these in such a way that the expression always evaluates to a certain type but realized that I wasn't really smart enough hahaha. Either that or it's not really possible, I didn't give that too much thought.

But then when I got further I came across a lot of properties such as [line-join](https://docs.mapbox.com/style-spec/reference/layers/#layout-line-line-join) that are actually enums. So I did try to pull those into the expressions but it came out a little bit wonky I think.. With the `Enum` inside the `enum`. Maybe they should just be strings after all? Anyway, please have a look and a think about it!

Irrespective of where we take the rest of the layer stuff, having a good representation of the mapbox expressions is quite nice so it makes sense to spend some time to find an approach that makes us happy!